database_build/db_build.sql

```diff
@@ -3,29 +3,52 @@ BEGIN;
 DROP TABLE IF EXISTS memories, users, media CASCADE;
 
 CREATE TABLE memories (
-  id          SERIAL        PRIMARY KEY,
+  id          SERIAL          PRIMARY KEY,
+  user_id     SERIAL          NOT NULL,
   heading     VARCHAR(50)     NOT NULL,
   likes       INTEGER,
-  rating      INTEGER,
+  avgRating   DECIMAL,
   visits      INTEGER,
+  tag         VARCHAR(50)     NOT NULL,
   memory      VARCHAR(500)    NOT NULL
 );
 
 CREATE TABLE users (
-  id          SERIAL        PRIMARY KEY,
-  username    VARCHAR(64)   UNIQUE NOT NULL,
-  password    VARCHAR(64)  NOT NULL,
-  email    VARCHAR(64)  NOT NULL
+  id          SERIAL          PRIMARY KEY,
+  username    VARCHAR(64)     UNIQUE NOT NULL,
+  password    VARCHAR(64)     NOT NULL,
+  email       VARCHAR(64)     NOT NULL
 );
 
 CREATE TABLE media (
-  text_only    TEXT[],
-  audio    TEXT[],
-  video    TEXT[],
-  image    TEXT[]
+  text_only   INTEGER[],
```

Comment:
Is sql, you describe relations between tables using foreign keys.

In this case, the relation is `belongs to`.

Ie a `memory_image` belongs to a `memory`.

So, you will have something like
```sql
CREATE TABLE memory_text (
    text varchar(100) NOT NULL,
    FOREIGN KEY (memory_id) REFERENCES memories(memory_id)
);
```
Not super hot on my sql so that might need tweeking, but thats the general idea